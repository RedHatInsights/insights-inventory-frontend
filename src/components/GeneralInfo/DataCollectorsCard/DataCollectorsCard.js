/* eslint-disable camelcase */
import React, { useCallback, useState } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import LoadingCard from '../LoadingCard';
import DateFormat from '@redhat-cloud-services/frontend-components/DateFormat';
import {
  ExpandableRowContent,
  Table /* data-codemods */,
  TableVariant,
  Tbody,
  Td,
  Th,
  Thead,
  Tr,
} from '@patternfly/react-table';
import { Flex, FlexItem } from '@patternfly/react-core';
import { getDefaultCollectors } from '../selectors/selectors';

const DataCollectorsCardCore = ({
  detailLoaded,
  collectors,
  entity,
  dataMapper,
}) => {
  const [expandedNames, setExpandedNames] = useState([]);
  const setExpanded = useCallback(
    (collector, isExpanding = true) =>
      setExpandedNames((prevExpanded) => {
        const otherExpandedNames = prevExpanded.filter(
          (r) => r !== collector.name
        );
        return isExpanding
          ? [...otherExpandedNames, collector.name]
          : otherExpandedNames;
      }),
    []
  );
  const isExpanded = (collector) => expandedNames.includes(collector.name);
  const data = collectors ?? getDefaultCollectors(entity);
  return (
    <LoadingCard title="Data collectors" isLoading={!detailLoaded}>
      <Table
        aria-label="Data collectors"
        variant={TableVariant.compact}
        borders={false}
      >
        <Thead>
          <Tr className="ins-c__no-border">
            <Th />
            <Th>Name</Th>
            <Th>Status</Th>
            <Th>Last upload</Th>
          </Tr>
        </Thead>
        {dataMapper
          ? dataMapper(data, isExpanded, setExpanded)
          : data.map((collector, rowIndex) => (
              <Tbody key={collector.name} isExpanded={isExpanded(collector)}>
                <Tr>
                  {collector.details.name ? (
                    <Td
                      expand={
                        collector.details
                          ? {
                              rowIndex,
                              isExpanded: isExpanded(collector),
                              onToggle: () =>
                                setExpanded(collector, !isExpanded(collector)),
                            }
                          : undefined
                      }
                      style={{ paddingLeft: 0 }}
                    />
                  ) : (
                    <Td />
                  )}
                  <Td dataLabel="Name">{collector.name}</Td>
                  <Td dataLabel="Status">{collector.status}</Td>
                  <Td dataLabel="Last upload">
                    {collector.updated ? (
                      <DateFormat date={collector.updated} type="exact" />
                    ) : (
                      'N/A'
                    )}
                  </Td>
                </Tr>
                {collector.details && collector.details.name && (
                  <Tr isExpanded={isExpanded(collector)}>
                    <Td />
                    <Td colSpan={3}>
                      <ExpandableRowContent>
                        <Flex>
                          <FlexItem className="ins-c__flex-row-margin">
                            {`${collector.details.name}:`}
                          </FlexItem>
                          <FlexItem grow={{ default: 'grow' }}>
                            {collector.details.id ?? 'N/A'}
                          </FlexItem>
                        </Flex>
                      </ExpandableRowContent>
                    </Td>
                  </Tr>
                )}
              </Tbody>
            ))}
      </Table>
    </LoadingCard>
  );
};

DataCollectorsCardCore.propTypes = {
  detailLoaded: PropTypes.bool,
  collectors: PropTypes.array,
  dataMapper: PropTypes.func,
  entity: PropTypes.shape({
    per_reporter_staleness: PropTypes.object,
  }),
};
DataCollectorsCardCore.defaultProps = {
  detailLoaded: false,
};

export const DataCollectorsCard = connect(
  ({ entityDetails: { entity }, systemProfileStore: { systemProfile } }) => ({
    entity,
    systemProfile,
    detailLoaded: systemProfile?.loaded,
    defaultCollectors: getDefaultCollectors(entity),
  })
)(DataCollectorsCardCore);

DataCollectorsCard.propTypes = DataCollectorsCardCore.propTypes;
DataCollectorsCard.defaultProps = DataCollectorsCardCore.defaultProps;

export default DataCollectorsCard;
