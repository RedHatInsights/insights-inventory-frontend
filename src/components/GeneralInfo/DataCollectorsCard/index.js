export { default } from './DataCollectorsCard';
export * from './DataCollectorsCard';
