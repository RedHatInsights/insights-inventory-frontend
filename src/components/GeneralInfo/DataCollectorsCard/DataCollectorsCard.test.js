/* eslint-disable camelcase */
import React from 'react';
import DataCollectorsCard from './DataCollectorsCard';
import configureStore from 'redux-mock-store';
import { render } from '@testing-library/react';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: 'localhost:3000/example/path',
  }),
  useHistory: () => ({
    push: () => undefined,
  }),
}));

describe('DataCollectorsCard', () => {
  let initialState;
  let mockStore;

  beforeEach(() => {
    mockStore = configureStore();
    initialState = {
      entityDetails: {
        entity: {
          per_reporter_staleness: {
            puptoo: {
              check_in_succeeded: true,
              last_check_in: '2022-05-13T07:42:21.663665+00:00',
              stale_timestamp: '2260-01-01T00:00:00+00:00',
            },
          },
          insights_id: '1234',
        },
      },
      systemProfileStore: {
        systemProfile: {
          loaded: true,
        },
      },
    };
  });

  it('should render correctly - no data', () => {
    const store = mockStore({ systemProfileStore: {}, entityDetails: {} });
    const view = render(<DataCollectorsCard store={store} />);
    expect(view.asFragment()).toMatchSnapshot();
  });

  it('should render correctly with data', () => {
    const store = mockStore(initialState);
    const view = render(<DataCollectorsCard store={store} />);
    expect(view.asFragment()).toMatchSnapshot();
  });

  it('should render custom collectors', () => {
    const store = mockStore(initialState);
    const view = render(
      <DataCollectorsCard
        store={store}
        {...{
          collectors: [
            {
              name: 'collector name',
              status: true,
              updated: '2260-01-01T00:00:00+00:00',
              details: {
                name: 'reporter id',
                id: '1234567',
              },
            },
          ],
        }}
      />
    );
    expect(view.asFragment()).toMatchSnapshot();
  });
});
