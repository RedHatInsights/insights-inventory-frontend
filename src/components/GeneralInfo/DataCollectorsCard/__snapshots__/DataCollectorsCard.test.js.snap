// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataCollectorsCard should render correctly - no data 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-1"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-1"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            Data collectors
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <table
          aria-label="Data collectors"
          class="pf-c-table pf-m-grid-md pf-m-compact pf-m-no-border-rows"
          data-ouia-component-id="OUIA-Generated-Table-1"
          data-ouia-component-type="PF4/Table"
          data-ouia-safe="true"
          role="grid"
        >
          <thead
            class=""
          >
            <tr
              aria-label=""
              class="ins-c__no-border"
              data-ouia-component-id="OUIA-Generated-TableRow-1"
              data-ouia-component-type="PF4/TableRow"
              data-ouia-safe="true"
            >
              <th
                class=""
              />
              <th
                class=""
                scope="col"
              >
                Name
              </th>
              <th
                class=""
                scope="col"
              >
                Status
              </th>
              <th
                class=""
                scope="col"
              >
                Last upload
              </th>
            </tr>
          </thead>
          <tbody
            class=""
            role="rowgroup"
          >
            <tr
              aria-label=""
              class=""
              data-ouia-component-id="OUIA-Generated-TableRow-2"
              data-ouia-component-type="PF4/TableRow"
              data-ouia-safe="true"
            >
              <td
                class="pf-c-table__toggle"
                style="padding-left:0"
              >
                <button
                  aria-disabled="false"
                  aria-expanded="false"
                  aria-label="Details"
                  aria-labelledby="simple-node0 expand-toggle0"
                  class="pf-c-button pf-m-plain"
                  data-ouia-component-id="OUIA-Generated-Button-plain-1"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  id="expand-toggle0"
                  type="button"
                >
                  <div
                    class="pf-c-table__toggle-icon"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align:-0.125em"
                      viewBox="0 0 320 512"
                      width="1em"
                    >
                      <path
                        d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
                      />
                    </svg>
                  </div>
                </button>
              </td>
              <td
                class=""
                data-label="Name"
              >
                insights-client
              </td>
              <td
                class=""
                data-label="Status"
              >
                N/A
              </td>
              <td
                class=""
                data-label="Last upload"
              >
                N/A
              </td>
            </tr>
            <tr
              aria-label=""
              class="pf-c-table__expandable-row"
              data-ouia-component-id="OUIA-Generated-TableRow-3"
              data-ouia-component-type="PF4/TableRow"
              data-ouia-safe="true"
              hidden=""
            >
              <td
                class=""
              />
              <td
                class=""
                colspan="3"
              >
                <div
                  class="pf-c-table__expandable-row-content"
                >
                  <div
                    class="pf-l-flex"
                  >
                    <div
                      class="ins-c__flex-row-margin"
                    >
                      Insights id:
                    </div>
                    <div
                      class="pf-m-grow"
                    >
                      N/A
                    </div>
                  </div>
                </div>
              </td>
            </tr>
          </tbody>
          <tbody
            class=""
            role="rowgroup"
          >
            <tr
              aria-label=""
              class=""
              data-ouia-component-id="OUIA-Generated-TableRow-4"
              data-ouia-component-type="PF4/TableRow"
              data-ouia-safe="true"
            >
              <td
                class="pf-c-table__toggle"
                style="padding-left:0"
              >
                <button
                  aria-disabled="false"
                  aria-expanded="false"
                  aria-label="Details"
                  aria-labelledby="simple-node1 expand-toggle1"
                  class="pf-c-button pf-m-plain"
                  data-ouia-component-id="OUIA-Generated-Button-plain-2"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  id="expand-toggle1"
                  type="button"
                >
                  <div
                    class="pf-c-table__toggle-icon"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align:-0.125em"
                      viewBox="0 0 320 512"
                      width="1em"
                    >
                      <path
                        d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
                      />
                    </svg>
                  </div>
                </button>
              </td>
              <td
                class=""
                data-label="Name"
              >
                subscription-manager
              </td>
              <td
                class=""
                data-label="Status"
              >
                N/A
              </td>
              <td
                class=""
                data-label="Last upload"
              >
                N/A
              </td>
            </tr>
            <tr
              aria-label=""
              class="pf-c-table__expandable-row"
              data-ouia-component-id="OUIA-Generated-TableRow-5"
              data-ouia-component-type="PF4/TableRow"
              data-ouia-safe="true"
              hidden=""
            >
              <td
                class=""
              />
              <td
                class=""
                colspan="3"
              >
                <div
                  class="pf-c-table__expandable-row-content"
                >
                  <div
                    class="pf-l-flex"
                  >
                    <div
                      class="ins-c__flex-row-margin"
                    >
                      Subscription manager id:
                    </div>
                    <div
                      class="pf-m-grow"
                    >
                      N/A
                    </div>
                  </div>
                </div>
              </td>
            </tr>
          </tbody>
          <tbody
            class=""
            role="rowgroup"
          >
            <tr
              aria-label=""
              class=""
              data-ouia-component-id="OUIA-Generated-TableRow-6"
              data-ouia-component-type="PF4/TableRow"
              data-ouia-safe="true"
            >
              <td
                class=""
              />
              <td
                class=""
                data-label="Name"
              >
                Satellite/Discovery
              </td>
              <td
                class=""
                data-label="Status"
              >
                N/A
              </td>
              <td
                class=""
                data-label="Last upload"
              >
                N/A
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</article>
`;

exports[`DataCollectorsCard should render correctly with data 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-2"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-2"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            Data collectors
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <table
          aria-label="Data collectors"
          class="pf-c-table pf-m-grid-md pf-m-compact pf-m-no-border-rows"
          data-ouia-component-id="OUIA-Generated-Table-2"
          data-ouia-component-type="PF4/Table"
          data-ouia-safe="true"
          role="grid"
        >
          <thead
            class=""
          >
            <tr
              aria-label=""
              class="ins-c__no-border"
              data-ouia-component-id="OUIA-Generated-TableRow-7"
              data-ouia-component-type="PF4/TableRow"
              data-ouia-safe="true"
            >
              <th
                class=""
              />
              <th
                class=""
                scope="col"
              >
                Name
              </th>
              <th
                class=""
                scope="col"
              >
                Status
              </th>
              <th
                class=""
                scope="col"
              >
                Last upload
              </th>
            </tr>
          </thead>
          <tbody
            class=""
            role="rowgroup"
          >
            <tr
              aria-label=""
              class=""
              data-ouia-component-id="OUIA-Generated-TableRow-8"
              data-ouia-component-type="PF4/TableRow"
              data-ouia-safe="true"
            >
              <td
                class="pf-c-table__toggle"
                style="padding-left:0"
              >
                <button
                  aria-disabled="false"
                  aria-expanded="false"
                  aria-label="Details"
                  aria-labelledby="simple-node0 expand-toggle0"
                  class="pf-c-button pf-m-plain"
                  data-ouia-component-id="OUIA-Generated-Button-plain-3"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  id="expand-toggle0"
                  type="button"
                >
                  <div
                    class="pf-c-table__toggle-icon"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align:-0.125em"
                      viewBox="0 0 320 512"
                      width="1em"
                    >
                      <path
                        d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
                      />
                    </svg>
                  </div>
                </button>
              </td>
              <td
                class=""
                data-label="Name"
              >
                insights-client
              </td>
              <td
                class=""
                data-label="Status"
              >
                Active
              </td>
              <td
                class=""
                data-label="Last upload"
              >
                13 May 2022 07:42 UTC
              </td>
            </tr>
            <tr
              aria-label=""
              class="pf-c-table__expandable-row"
              data-ouia-component-id="OUIA-Generated-TableRow-9"
              data-ouia-component-type="PF4/TableRow"
              data-ouia-safe="true"
              hidden=""
            >
              <td
                class=""
              />
              <td
                class=""
                colspan="3"
              >
                <div
                  class="pf-c-table__expandable-row-content"
                >
                  <div
                    class="pf-l-flex"
                  >
                    <div
                      class="ins-c__flex-row-margin"
                    >
                      Insights id:
                    </div>
                    <div
                      class="pf-m-grow"
                    >
                      1234
                    </div>
                  </div>
                </div>
              </td>
            </tr>
          </tbody>
          <tbody
            class=""
            role="rowgroup"
          >
            <tr
              aria-label=""
              class=""
              data-ouia-component-id="OUIA-Generated-TableRow-10"
              data-ouia-component-type="PF4/TableRow"
              data-ouia-safe="true"
            >
              <td
                class="pf-c-table__toggle"
                style="padding-left:0"
              >
                <button
                  aria-disabled="false"
                  aria-expanded="false"
                  aria-label="Details"
                  aria-labelledby="simple-node1 expand-toggle1"
                  class="pf-c-button pf-m-plain"
                  data-ouia-component-id="OUIA-Generated-Button-plain-4"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  id="expand-toggle1"
                  type="button"
                >
                  <div
                    class="pf-c-table__toggle-icon"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align:-0.125em"
                      viewBox="0 0 320 512"
                      width="1em"
                    >
                      <path
                        d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
                      />
                    </svg>
                  </div>
                </button>
              </td>
              <td
                class=""
                data-label="Name"
              >
                subscription-manager
              </td>
              <td
                class=""
                data-label="Status"
              >
                N/A
              </td>
              <td
                class=""
                data-label="Last upload"
              >
                N/A
              </td>
            </tr>
            <tr
              aria-label=""
              class="pf-c-table__expandable-row"
              data-ouia-component-id="OUIA-Generated-TableRow-11"
              data-ouia-component-type="PF4/TableRow"
              data-ouia-safe="true"
              hidden=""
            >
              <td
                class=""
              />
              <td
                class=""
                colspan="3"
              >
                <div
                  class="pf-c-table__expandable-row-content"
                >
                  <div
                    class="pf-l-flex"
                  >
                    <div
                      class="ins-c__flex-row-margin"
                    >
                      Subscription manager id:
                    </div>
                    <div
                      class="pf-m-grow"
                    >
                      N/A
                    </div>
                  </div>
                </div>
              </td>
            </tr>
          </tbody>
          <tbody
            class=""
            role="rowgroup"
          >
            <tr
              aria-label=""
              class=""
              data-ouia-component-id="OUIA-Generated-TableRow-12"
              data-ouia-component-type="PF4/TableRow"
              data-ouia-safe="true"
            >
              <td
                class=""
              />
              <td
                class=""
                data-label="Name"
              >
                Satellite/Discovery
              </td>
              <td
                class=""
                data-label="Status"
              >
                N/A
              </td>
              <td
                class=""
                data-label="Last upload"
              >
                N/A
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</article>
`;

exports[`DataCollectorsCard should render custom collectors 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-3"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-3"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            Data collectors
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <table
          aria-label="Data collectors"
          class="pf-c-table pf-m-grid-md pf-m-compact pf-m-no-border-rows"
          data-ouia-component-id="OUIA-Generated-Table-3"
          data-ouia-component-type="PF4/Table"
          data-ouia-safe="true"
          role="grid"
        >
          <thead
            class=""
          >
            <tr
              aria-label=""
              class="ins-c__no-border"
              data-ouia-component-id="OUIA-Generated-TableRow-13"
              data-ouia-component-type="PF4/TableRow"
              data-ouia-safe="true"
            >
              <th
                class=""
              />
              <th
                class=""
                scope="col"
              >
                Name
              </th>
              <th
                class=""
                scope="col"
              >
                Status
              </th>
              <th
                class=""
                scope="col"
              >
                Last upload
              </th>
            </tr>
          </thead>
          <tbody
            class=""
            role="rowgroup"
          >
            <tr
              aria-label=""
              class=""
              data-ouia-component-id="OUIA-Generated-TableRow-14"
              data-ouia-component-type="PF4/TableRow"
              data-ouia-safe="true"
            >
              <td
                class="pf-c-table__toggle"
                style="padding-left:0"
              >
                <button
                  aria-disabled="false"
                  aria-expanded="false"
                  aria-label="Details"
                  aria-labelledby="simple-node0 expand-toggle0"
                  class="pf-c-button pf-m-plain"
                  data-ouia-component-id="OUIA-Generated-Button-plain-5"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  id="expand-toggle0"
                  type="button"
                >
                  <div
                    class="pf-c-table__toggle-icon"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align:-0.125em"
                      viewBox="0 0 320 512"
                      width="1em"
                    >
                      <path
                        d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
                      />
                    </svg>
                  </div>
                </button>
              </td>
              <td
                class=""
                data-label="Name"
              >
                collector name
              </td>
              <td
                class=""
                data-label="Status"
              />
              <td
                class=""
                data-label="Last upload"
              >
                01 Jan 2260 00:00 UTC
              </td>
            </tr>
            <tr
              aria-label=""
              class="pf-c-table__expandable-row"
              data-ouia-component-id="OUIA-Generated-TableRow-15"
              data-ouia-component-type="PF4/TableRow"
              data-ouia-safe="true"
              hidden=""
            >
              <td
                class=""
              />
              <td
                class=""
                colspan="3"
              >
                <div
                  class="pf-c-table__expandable-row-content"
                >
                  <div
                    class="pf-l-flex"
                  >
                    <div
                      class="ins-c__flex-row-margin"
                    >
                      reporter id:
                    </div>
                    <div
                      class="pf-m-grow"
                    >
                      1234567
                    </div>
                  </div>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</article>
`;
